src/main/java/gift/service/MemberServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package gift.service;
+
+import gift.dto.LoginResponseDto;
+import gift.dto.MemberRequestDto;
+import gift.entity.Member;
+import gift.repository.MemberRepository;
+import gift.util.JwtUtil;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+
+@Service
+public class MemberServiceImpl implements MemberService {
+    private final MemberRepository memberRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final JwtUtil jwtUtil;
+
+    public MemberServiceImpl(MemberRepository memberRepository, PasswordEncoder passwordEncoder, JwtUtil jwtUtil) {
+        this.memberRepository = memberRepository;
+        this.passwordEncoder = passwordEncoder;
+        this.jwtUtil = jwtUtil;
+    }
+
+    @Override
+    @Transactional
+    public LoginResponseDto saveMember(MemberRequestDto dto) {
+        if (memberRepository.existsByEmail(dto.email())) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "이미 존재하는 이메일입니다.");
+        }
+
+        String encodedPassword = passwordEncoder.encode(dto.password());
+        Member member = new Member(dto.email(), encodedPassword);
+        Member savedMember = memberRepository.saveMember(member);
+
+        String token = jwtUtil.generateToken(savedMember.getEmail(), savedMember.getId(), savedMember.getRole());
+        return new LoginResponseDto(token);
+    }
+
+    @Override
+    @Transactional
+    public LoginResponseDto loginMember(MemberRequestDto dto) {
+        Member member = memberRepository.findByEmail(dto.email())
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.FORBIDDEN, "잘못된 이메일입니다."));
+
+        if (!passwordEncoder.matches(dto.password(), member.getPassword())) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN, "잘못된 비밀번호입니다.");
```

Comment:
```suggestion
            throw new ResponseStatusException(HttpStatus.FORBIDDEN, "이메일 또는 비밀번호가 일치하지 않습니다.");
```
- 보안과 관련한 예외 메시지는 구체적으로 작성하면 Brute Force 등의 공격을 허용하기 좋습니다.
- HttpStatus와 같이 Presentation Layer에 해당하는 요소와 결합되어 있네요~